client/app/telecom/pack/xdsl/modem/service/pack-xdsl-modem-service.controller.js

```diff
@@ -0,0 +1,331 @@
+angular.module('managerApp').controller('XdslModemServiceCtrl', class XdslModemServiceCtrl {
+  /* @ngInject */
+  constructor($q, $stateParams, $translate, $uibModal, OvhApiXdsl, TucPackXdslModemMediator,
+    TucToast) {
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.$q = $q;
+    this.OvhApiXdsl = OvhApiXdsl;
+    this.TucToast = TucToast;
+    this.$uibModal = $uibModal;
+    this.mediator = TucPackXdslModemMediator;
+
+    this.service = {};
+  }
+
+  getService() {
+    this.getBlocIP();
+    this.getCallWaiting();
+    this.getContentSharing();
+    this.getFTP();
+    this.getIpsecAlg();
+    this.getSipAlg();
+    this.getUPnP();
+  }
+
+  getBlocIP() {
+    return this.OvhApiXdsl.Modem().BlocIp().v6().get({
+      xdslId: this.$stateParams.serviceName,
+    }).$promise.then(({ xdslTask }) => {
+      if (xdslTask === 'enabled') {
+        this.service.blocIP = true;
+      } else {
+        this.service.blocIP = false;
+      }
+      this.service.blocIPTmp = this.service.blocIP;
+      this.service.isBlocIP = true;
+      return xdslTask;
+    }).catch(() => {
+      this.service.isBlocIP = false;
+    });
+  }
+
+  changeBlocIp(value) {
+    this.updateBlocIp(value ? 'enabled' : 'disabled');
+  }
+
+  updateBlocIp(status) {
+    return this.OvhApiXdsl.Modem().BlocIp().v6().post({
+      xdslId: this.$stateParams.serviceName,
+    },
+    {
+      status,
+    }).$promise.then((xdslTask) => {
+      this.mediator.setTask('changeModemBlocIpStatus');
+      this.service.blocIPTmp = this.service.blocIP;
```

Comment:
Now there is a `blocIPTmp ` ? What's the difference with `blocIP` and `isBlocIP` ?

Reply:
For the modem configuration, it's possible that the BlocIp is not available (this is why I defined isBlocIp). The blocIp is the value set  from the API. And the blocIpTmp is to get the previous value if there is an error with update action